ddl/ddl.go

```diff
@@ -201,28 +198,39 @@ type DDL interface {
 
 // ddl represents the statements which are used to define the database structure or schema.
 type ddl struct {
-	m sync.RWMutex
+	m          sync.RWMutex
+	infoHandle *infoschema.Handle
+	quitCh     chan struct{}
 
-	infoHandle   *infoschema.Handle
-	hook         Callback
-	hookMu       sync.RWMutex
+	*ddlCtx
+	workers []*worker
+}
+
+// ddlCtx is the context when we use worker to handle DDL jobs.
+type ddlCtx struct {
+	uuid         string
 	store        kv.Storage
 	ownerManager owner.Manager
 	schemaSyncer SchemaSyncer
-	// lease is schema seconds.
-	lease        time.Duration
-	uuid         string
 	ddlJobCh     chan struct{}
 	ddlJobDoneCh chan struct{}
 	ddlEventCh   chan<- *util.Event
-	// reorgCtx is for reorganization.
-	reorgCtx *reorgCtx
+	lease        time.Duration // lease is schema lease.
+
+	// hook may be modified.
+	hook   Callback
+	hookMu sync.RWMutex
 
-	quitCh chan struct{}
-	wait   sync.WaitGroup
+	workerVars *variable.SessionVars // workerVars is used for Binlog.
```

Comment:
Could we put binlog client here directly?

Reply:
I think we can. But this is original code, do we need to update in this PR?